api/query/atoms.go

```diff
@@ -906,6 +930,26 @@ func (l *AbstractLink) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// UnmarshalJSON for AbstractTestLabel attempts to interpret a query atom as
+// {"testlabel":<label string>}.
+func (t *AbstractTestLabel) UnmarshalJSON(b []byte) error {
+	var data map[string]*json.RawMessage
+	if err := json.Unmarshal(b, &data); err != nil {
+		return err
+	}
+	labelMsg, ok := data["testlabel"]
+	if !ok {
+		return errors.New(`Missing testlabel pattern property: "testlabel"`)
+	}
+	var label string
+	if err := json.Unmarshal(*labelMsg, &label); err != nil {
+		return errors.New(`Missing testlabel pattern property "label" is not a string`)
```

Comment:
This says "label" and not "testlabel", is that intentional?

Reply:
Yup it is intentional. It is different from checking if `labelMsg, ok := data["testlabel"]` key words exist. The `label` here is the label of a metadata in string format.

Reply:
Yup you are right. The original wording was wrong because nothing was `missing` there. Thanks for the correction!